x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rest/ServerSentEventsRestActionListener.java

```diff
@@ -115,6 +121,12 @@ private void initializeStream(InferenceAction.Response response) {
                 )
             );
         });
+
+        nextBodyPartListener = ContextPreservingActionListener.wrapPreservingContext(
```

Comment:
When I manually test this, it fixes the issue. I always get the header back as expected. Without the fix the `_stream` api usually does not return the header. I believe this is because we are create a new listener that maybe gets executed with a different thread context. I think if we responded in the `onResponse` implementation of this class it'd probably avoid the issue. But that's not how this was designed so I'm keeping the structure as it was.